src/js/dp/release-calendar.js

```diff
@@ -30,3 +36,118 @@ document.addEventListener("DOMContentLoaded", function () {
+      ".ons-radio__input[type=radio]:not(input:disabled)"
+    ),
+  ].map((topFilter) => {
+    topFilter.addEventListener("change", async (e) => {
+      gtmDataLayerPush({
+        event: "Filter",
+        "filter-by": e.target.name,
+        selected: e.target.value.replace("type-", ""),
+      });
+    });
+  });
+
+  document.querySelector('[aria-label="Keywords"]').onsubmit = async (e) => {
+    const formData = new FormData(e.target);
+    const formProps = Object.fromEntries(formData);
+    gtmDataLayerPush({
+      event: "Filter",
+      "filter-by": "search",
+      "search-term": formProps.keywords,
+    });
+  };
+
+  document.querySelector('[aria-label="Released date"]').onsubmit = async (
+    e
+  ) => {
+    const formData = new FormData(e.target);
+    const formProps = Object.fromEntries(formData);
+    const startDate = `${formProps["after-year"]}/${formProps["after-month"]}/${formProps["after-day"]}`;
+    const endDate = `${formProps["before-year"]}/${formProps["before-month"]}/${formProps["before-day"]}`;
+
+    const numberOfDays = days(
+      new Date(
+        formProps["after-year"],
+        formProps["after-month"],
+        formProps["after-day"]
+      ),
+      new Date(
+        formProps["before-year"],
+        formProps["before-month"],
+        formProps["before-day"]
+      )
+    );
+
+    gtmDataLayerPush({
+      event: "Filter",
+      "filter-by": "date-range",
+      "start-date": startDate,
+      "end-date": endDate,
+      "number-of-days": numberOfDays,
+    });
+  };
+}
+
+if (releasePageContainer) {
+  const releaseStatus = releasePageContainer.getAttribute(
+    "data-gtm-release-status"
+  );
+  const releaseDate = new Date(
+    releasePageContainer.getAttribute("data-gtm-release-date")
+  );
+  const releaseTime = new Date(
+    releasePageContainer.getAttribute("data-gtm-release-time")
+  );
+  const releaseDateStatus = releasePageContainer.getAttribute(
+    "data-gtm-release-date-status"
+  );
+  const nextReleaseDate = new Date(
+    releasePageContainer.getAttribute("data-gtm-next-release-date")
+  );
+  const contactName = releasePageContainer.getAttribute(
+    "data-gtm-contact-name"
+  );
+
+  const year = releaseDate.getFullYear().toString().padStart(2, "0");
+  const month = releaseDate.getMonth().toString().padStart(2, "0");
+  const date = releaseDate.getDate().toString().padStart(2, "0");
+  const hour = releaseTime.getHours().toString().padStart(2, "0");
+  const minutes = releaseTime.getMinutes().toString().padStart(2, "0");
+
+  const nextReleaseYear = nextReleaseDate
+    .getFullYear()
+    .toString()
+    .padStart(2, "0");
+  const nextReleaseMonth = nextReleaseDate
+    .getMonth()
+    .toString()
+    .padStart(2, "0");
+  const nextReleaseDayDate = nextReleaseDate
+    .getDate()
+    .toString()
+    .padStart(2, "0");
+
+  gtmDataLayerPush({
+    "release-status": releaseStatus,
+    "release-date": `${year}${month}${date}`,
```

Comment:
No delimiters like `/` or `-` expected in the date e.g. `06/07/2011`?

Reply:
base on the date formart on the ticket, we dont need delimiter 
https://trello.com/c/u6qUgeik/977-release-pages-datalayer-elements